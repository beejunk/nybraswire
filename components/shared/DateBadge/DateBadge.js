// @flow

import React, { useContext } from 'react';
import { getMonthStr, appendSuffixToDate, getFormattedTime } from '../../../utils/dateUtils';
import ThemeContext from '../../../theme';

type Props = {
  timestamp: number,
};

const DateBadge = (props: Props) => {
  const { timestamp } = props;
  const { colors } = useContext(ThemeContext);
  const date = new Date(timestamp);
  const dayOfMonth = appendSuffixToDate(date.getDate());
  const month = getMonthStr(date.getMonth());
  const year = date.getFullYear();
  const time = getFormattedTime(date);

  // TODO: Why is minWidth necessary here?
  return (
    <div
      className="DateBadge"
      css={{
        borderRadius: '5px',
        backgroundColor: colors.gray900,
        display: 'flex',
        color: '#fff',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',
        lineHeight: '1.25rem',
        width: '5rem',
        height: '5rem',
        minWidth: '5rem',
      }}
    >
      <div css={{ fontSize: '0.5rem', lineHeight: '0.5rem' }}>{year}</div>
      <div css={{ fontWeight: 700 }}>{month}</div>
      <div>{dayOfMonth}</div>
      <div css={{ fontSize: '0.5rem', lineHeight: '0.75rem' }}>{time}</div>
    </div>
  );
};

export default DateBadge;
