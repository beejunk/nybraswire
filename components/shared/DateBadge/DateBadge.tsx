import React, { useContext, useEffect, useState } from 'react';

import {
  getMonthStr,
  appendSuffixToDate,
  getFormattedTime,
} from '../../../utils/dateUtils';
import ThemeContext from '../../../theme';

type Props = {
  timestamp: number;
};

const DateBadge: React.FunctionComponent<Props> = function DateBadge(props) {
  const { timestamp } = props;
  const { colors } = useContext(ThemeContext);
  const date = new Date(timestamp);
  const dayOfMonth = appendSuffixToDate(date.getDate());
  const month = getMonthStr(date.getMonth());
  const year = date.getFullYear();
  const time = getFormattedTime(date);
  const [hasMounted, setHasMounted] = useState(false);

  useEffect(() => {
    setHasMounted(true);
  }, []);

  // TODO: Why is minWidth necessary here?
  return (
    <div
      className="DateBadge"
      css={{
        borderRadius: '5px',
        backgroundColor: colors.gray900,
        display: 'flex',
        color: '#fff',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',
        lineHeight: '1.25rem',
        width: '5rem',
        height: '5rem',
        minWidth: '5rem',
      }}
    >
      {/* Render date client-side only to avoid differences between server/client */}
      {hasMounted && (
        <>
          <div css={{ fontSize: '0.5rem', lineHeight: '0.5rem' }}>{year}</div>
          <div css={{ fontWeight: 700 }}>{month}</div>
          <div>{dayOfMonth}</div>
          <div css={{ fontSize: '0.5rem', lineHeight: '0.75rem' }}>{time}</div>
        </>
      )}
    </div>
  );
};

export default DateBadge;
