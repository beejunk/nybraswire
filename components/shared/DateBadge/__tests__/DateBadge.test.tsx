import { shallow } from 'enzyme';
import DateBadge from '../DateBadge';

describe('DateBadge', () => {
  let wrapper;

  beforeEach(() => {
    const date = new Date('2019-08-10T17:34:00Z');
    wrapper = shallow(<DateBadge timestamp={date.getTime()} />);
  });

  it('should pass snapshot test', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
