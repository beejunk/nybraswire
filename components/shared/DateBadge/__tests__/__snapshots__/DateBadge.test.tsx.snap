// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateBadge should pass snapshot test 1`] = `"<div class=\\"DateBadge css-1p72s2n-DateBadge\\"><div class=\\"css-1l7x69b-DateBadge\\">2019</div><div class=\\"css-1t6okqr-DateBadge\\">August</div><div>10th</div><div class=\\"css-7y2pl5-DateBadge\\">0:34 pm</div></div>"`;
